### PR TITLE
Update OAUTH connection options and username is no longer required.

### PR DESCRIPTION
**Pull request by sfc-gh-ext-simba-ll**
Issue [464](https://github.com/snowflakedb/snowflake-sdks-drivers-issues-teamwork/issues/464)

### CODE REVIEW

File: lib/connection/connection_config.js [hunk 1]
```diff
@@ -145,13 +145,18 @@ function ConnectionConfig(options, validateCredentials, qaMode, clientInfo)
...
+    {
+      // check for missing username
+      Errors.checkArgumentExists(Util.exists(options.username),
+        ErrorCodes.ERR_CONN_CREATE_MISSING_USERNAME);
 
-    // check for invalid username
-    Errors.checkArgumentValid(Util.isString(options.username),
-      ErrorCodes.ERR_CONN_CREATE_INVALID_USERNAME);
+      // check for invalid username
+      Errors.checkArgumentValid(Util.isString(options.username),
```
Comment (sfc-gh-igarish) [inline lib/connection/connection_config.js:157]: If user is provided in OAUTH authentication then needs to validate it. No?

**Comment by sfc-gh-igarish** [comment]
Changes looks good, but it has test failures.

**Comment by sfc-gh-ext-simba-ll** [comment]
> Changes looks good, but it has test failures.

There is a new vulnerability not related to this PR. Louie is looking into that issue.

**Comment by sfc-gh-dprzybysz** [comment]
> > Changes looks good, but it has test failures.
> 
> There is a new vulnerability not related to this PR. Louie is looking into that issue.

I merged the quick fix to enable the build: https://github.com/snowflakedb/snowflake-connector-nodejs/pull/542
You can rebase and it should be fine

File: lib/connection/connection_config.js [hunk 2]
```diff
@@ -145,13 +145,21 @@ function ConnectionConfig(options, validateCredentials, qaMode, clientInfo)
...
+        ErrorCodes.ERR_CONN_CREATE_MISSING_USERNAME);
+    }
 
-    // check for invalid username
-    Errors.checkArgumentValid(Util.isString(options.username),
-      ErrorCodes.ERR_CONN_CREATE_INVALID_USERNAME);
+    if (Util.exists(options.username))
+    {
+      // check for invalid username
+      Errors.checkArgumentValid(Util.isString(options.username),
```
Comment (sfc-gh-dprzybysz) [inline lib/connection/connection_config.js:160]: can we add this check in the newly added if above? the `if` body above checks user name existence and we don't need additional `if`. Additionally, in the same way it's done for password check below only for default authenticator

File: test/unit/connection/connection_config_test.js [hunk 3]
```diff
@@ -721,6 +835,20 @@ describe('ConnectionConfig: basic', function ()
           username: 'username',
           password: 'password'
         }
+      },
+      {
+        name: 'oauthwithoutusername',
```
Comment (sfc-gh-dprzybysz) [inline test/unit/connection/connection_config_test.js:840]: Can we added spaces as in other test names?

**Review by sfc-gh-dprzybysz — CHANGES_REQUESTED** [review]

File: lib/connection/connection_config.js (hunk 2 above)
Comment (sfc-gh-ext-simba-ll) [reply lib/connection/connection_config.js:160]: That's actually my initial fix. But as Ilesh mentioned, we should also validate username for OAUTH if username exist.

File: test/unit/connection/connection_config_test.js (hunk 3 above)
Comment (sfc-gh-ext-simba-ll) [reply test/unit/connection/connection_config_test.js]: updated

**Comment by sfc-gh-ext-simba-ll** [comment]
@sfc-gh-igarish @sfc-gh-dprzybysz 
Tests succeeded, once you approve it. I'll merge.

**Review by sfc-gh-dprzybysz — APPROVED** [review]

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM

**Pull request merged**